stellar/signer/main.go

```diff
@@ -0,0 +1,36 @@
+package signer
+
+import (
+	"encoding/hex"
+
+	"github.com/stellar/go/keypair"
+	"github.com/stellar/go/network"
+	"github.com/stellar/go/support/errors"
+	"github.com/stellar/go/xdr"
+)
+
+type Signer struct {
+	NetworkPassphrase string
+	SecretKey         string
+
+	kp *keypair.Full
+}
+
+// Sign signs an envelope.
+func (s *Signer) Sign(envelope xdr.TransactionEnvelope) (string, xdr.DecoratedSignature, error) {
+	hash, err := network.HashTransactionInEnvelope(envelope, s.NetworkPassphrase)
+	if err != nil {
+		return "", xdr.DecoratedSignature{}, errors.Wrap(err, "failed to hash transaction")
+	}
+
+	if s.kp == nil {
+		s.kp = keypair.MustParseFull(s.SecretKey)
```

Comment:
💡 You should move this into the Signer, make it a `*keypair.Full`, so that you know at the time of setting up the `Signer` whether it is valid or not, and then you don't have to have panics here in the code on an invalid value.